demonstrations/tutorial_lcu_blockencoding.py

```diff
@@ -0,0 +1,264 @@
+# a block encoding circuit with four unitaries:
+#
+# |
+#
+# .. figure:: ../demonstrations/lcu_blockencoding/thumbnail_lcu_blockencoding.png
+#     :align: center
+#     :width: 50%
+#     :target: javascript:void(0)
+#
+# |
+#
+# The reason for this name is that if we write down
+#  as a matrix, the operator :math:`A` is encoded inside a block of :math:`U`
+# defined by the subspace of all states where the auxiliary qubits are in state :math:`|0\rangle`.
+#
+# PennyLane supports direct implementation of `prepare <https://docs.pennylane.ai/en/latest/code/api/pennylane.StatePrep.html>`_
+# and `select <https://docs.pennylane.ai/en/latest/code/api/pennylane.Select.html?highlight=select>`_
+# operators. We'll go through them individually and use them to construct a block encoding circuit.
+# Prepare circuits can be constructed using the :func:`~.pennylane.StatePrep` operation, which takes
+# the normalized target state as input:
+
+dev1 = qml.device("default.qubit", wires=1)
+
+# normalized square roots of coefficients
+alphas = (np.sqrt(LCU.coeffs) / np.linalg.norm(np.sqrt(LCU.coeffs)))
+
+
+@qml.qnode(dev1)
+def prep_circuit():
+    qml.StatePrep(alphas, wires=0)
+    return qml.state()
+
+
+print("Target state: ", alphas)
+print("Output state: ", np.real(prep_circuit()))
+
+##############################################################################
+# Similarly, select circuits can be implemented using :func:`~.pennylane.Select`, which takes the
+# target unitaries as input. We specify the control wires directly, but the system wires are inherited
+# from the unitaries. Since :func:`~.pennylane.pauli_decompose` uses a canonical wire ordering, we
+# first map the wires to those used for the system register in our circuit:
+#
+
+dev2 = qml.device("default.qubit", wires=3)
+
+# unitaries
+ops = LCU.ops
+# relabeling wires 0 --> 1, and 1 --> 2
+unitaries = [qml.map_wires(op, {0: 1, 1: 2}) for op in ops]
+
+
+@qml.qnode(dev2)
+def sel_circuit(state):
+    qml.BasisState(state, wires=0)
+    qml.Select(unitaries, control=0)
+    return qml.expval(qml.PauliZ(2))
+
+
+# Select flips the last qubit if control is |1>
+print(sel_circuit([0]), sel_circuit([1]))
+
+##############################################################################
+# We can now combine these to construct a full LCU circuit. Here we make use of :fun:`~.pennylane.adjoint`
+# as a convenient way to invert the prepare circuit. We have chosen an input matrix that is already
+# normalized, so it can be seen appearing directly in the top-left block of the unitary describing
+# the full circuit --- the mark of a successful block encoding.
+
+
+@qml.qnode(dev2)
+def lcu_circuit():  # block_encode
+    # PREP
+    qml.StatePrep(alphas, wires=0)
+
+    # SEL
+    qml.Select(unitaries, control=0)
+
+    # PREP_dagger
+    qml.adjoint(qml.StatePrep(alphas, wires=0))
+    return qml.state()
+
+
+output_matrix = qml.matrix(lcu_circuit)()
+print(np.real(np.round(output_matrix)))
+print(A)
```

Comment:
For a cleaner print output. Also, as it displays currently in the build, the example is wrong and needs to be fixed.

```suggestion
output_matrix = qml.matrix(lcu_circuit)()
print("A:\n", A, "\n")
print("Block-encoded A:\n")
print(np.real(np.round(output_matrix)))
```